src/services/state-manager/state-manager.service.ts

```diff
@@ -1,39 +1,58 @@
-import _ from "lodash"
-import { Document } from "mongoose"
-import { IState } from "../../types/global/state"
-import states from "../../data/models/state-schema"
+import _ from "lodash";
+import mongoose from "mongoose";
+import { IState } from "../../types/global/state";
+import { StateNamesEnum } from "../../enums/state-names.enum";
 
 export class StateManagerService {
-  private static _instance: StateManagerService
+  private static _instance: StateManagerService;
 
-  public static getInstance() {
+  public static getInstance(): StateManagerService {
     if (_.isNil(StateManagerService._instance)) {
-      StateManagerService._instance = new StateManagerService()
+      StateManagerService._instance = new StateManagerService();
     }
-    return StateManagerService._instance
+    return StateManagerService._instance;
   }
 
-  public async getBotState(memberId: string): Promise<Document | void> {
-    return await states.findOne({ memberId }, (err: Error, stateFound: Document): IState => {
-      if (stateFound) return stateFound.schema.obj
-    })
-    .catch((err): void => {
-      console.log(`Error: State not found because: ${err}`)
-      return
-    })
+  private _currentState: IState = {
+    memberId: `None`,
+    state: {
+      data: ``,
+      name: StateNamesEnum.NORMAL,
+      step: 0,
+    },
+  };
+
+  public async getBotState(_memberId: string): Promise<IState> {
+    return new Promise((resolve, reject): void => {
+      void mongoose
```

Comment:
Same as above, I think you could return directly a promise.

Reply:
What? Isn't this what I'm doing?

Reply:
@asmodee-bellanie Like I said on another comment, I think you changed the way you handle mongoose and now the real return type is a Promise.
Let's try something KISS shall we?

- Remove `new Promise((resolve, reject): void => { void`.
- Change the resolve(...) to return Promise.resolve(...)
- Same for error

Reply:
Ok, I will do that.